CodeGen/Generators/UnitsNetGen/QuantityGenerator.cs

```diff
@@ -396,7 +385,34 @@ public static string GetAbbreviation({_unitEnumName} unit)
         /// <param name=""provider"">Format to use for localization. Defaults to <see cref=""CultureInfo.CurrentCulture"" /> if null.</param>
         public static string GetAbbreviation({_unitEnumName} unit, IFormatProvider? provider)
         {{
-            return UnitAbbreviationsCache.Default.GetDefaultAbbreviation(unit, provider);
+            return GetAbbreviations(unit, provider as CultureInfo).FirstOrDefault() ?? string.Empty;
+        }}
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name=""unit""></param>
+        /// <param name=""culture""></param>
+        /// <returns></returns>
+        public static IReadOnlyList<string> GetAbbreviations({_unitEnumName} unit, CultureInfo? culture = null)
+        {{
+            culture ??= CultureInfo.CurrentCulture;
+
+            if(!Abbreviations.TryGetValue((culture, unit), out var abbreviations))
+            {{
+                abbreviations = new List<string>();
+
+                const string resourceName = $""UnitsNet.GeneratedCode.Resources.{_quantity.Name}"";
+                var resourceManager = new ResourceManager(resourceName, typeof({_quantity.Name}).Assembly);
```

Comment:
I'm not familiar with manually instantiating `ResourceManager`. Should we cache and reuse these instances?

Reply:
Generally though, I think you are on to something good about reducing the generated code and moving things like localization to files that are read at runtime. 

The JIT clearly has issues with our codebase. Not sure if it is the large number of types/members or if it is just the large amount of code, or what exactly.

Reply:
Moving things into `QuantityInfo` is another way to avoid duplicating methods N times for N quantities.

The biggest hit is M members for M units though, which is roughly 10x the number of quantities.

Reply:
@angularsen I am doing it lazily to load into a List. However there is some caching done internally, if my read of the docs is correct.